pkg/segment/structs/evaluationstructs.go

```diff
@@ -1710,7 +1710,44 @@ func (self *TextExpr) EvaluateText(fieldToValue map[string]utils.CValueEnclosure
 	case "getfields":
 		fallthrough
 	case "typeof":
-		return "", fmt.Errorf("TextExpr.EvaluateText: dose not support functions:%v: right now", self.Op)
+		var val utils.CValueEnclosure
+
+		if self.Val.NumericExpr == nil {
+			val.Dtype = utils.SS_INVALID
+		} else {
+			val, ok := fieldToValue[self.Val.NumericExpr.Value] // The parser always sets the fieldName in NumericExpr even if it's a string
+			if !ok {
+				val.Dtype = utils.SS_INVALID
+			}
+		}
```

Comment:
I'm a little confused. The `Dtype` is `SS_INVALID` except for some cases when `NumericExpr != nil`, and in those cases wouldn't the `Dtype` be some numeric type? But later you have a `switch` that also checks for boolean and null types. When would those cases run?

Reply:
The parser sets the fieldName in `NumericExpr` but the `Dtype` can be string or boolean or null types as well (Any of these [link](https://github.com/venkateshamatam/siglens/blob/21eb99b433a9808fa0553b155191fc6f1a6002e6/pkg/segment/utils/segconsts.go#L141)) The switch cases are executed when the query does not have a fieldName like `typeOf(city)` but it has a raw value like `typeOf(17)` or `typeOf("somestring")` then the `Dtype` would be set as invalid and we determine the datatype of the raw value in the query using the logic in the switch cases below.